Analysis/Tasks/PWGHF/HFCandidateCreatorCascade.cxx

```diff
@@ -0,0 +1,304 @@
+                       errorDecayLength, errorDecayLengthXY,
+                       chi2PCA,
+                       pVecBach[0], pVecBach[1], pVecBach[2],
+                       pVecV0[0], pVecV0[1], pVecV0[2],
+                       impactParameterBach.getY(), impactParameterV0.getY(),
+                       std::sqrt(impactParameterBach.getSigmaY2()), std::sqrt(impactParameterV0.getSigmaY2()),
+                       casc.index0Id(), casc.indexV0Id(),
+                       casc.hfflag(),
+                       v0.x(), v0.y(), v0.z(),
+                       //v0.posTrack(), v0.negTrack(), // why this was not fine?
+                       v0.posTrack_as<MyTracks>().globalIndex(), v0.negTrack_as<MyTracks>().globalIndex(),
+                       v0.pxpos(), v0.pypos(), v0.pzpos(),
+                       v0.pxneg(), v0.pyneg(), v0.pzneg(),
+                       v0.dcaV0daughters(),
+                       v0.dcapostopv(),
+                       v0.dcanegtopv());
+
+      // fill histograms
+      if (b_doValPlots) {
+        // calculate invariant masses
+        auto arrayMomenta = array{pVecBach, pVecV0};
+        mass2K0sP = RecoDecay::M(arrayMomenta, array{massP, massK0s});
+        hmass2->Fill(mass2K0sP);
+      }
+    }
+  }
+};
+
+/// Extends the base table with expression columns.
+struct HFCandidateCreatorCascadeExpressions {
+  Spawns<aod::HfCandCascExt> rowCandidateCasc;
+  void init(InitContext const&) {}
+};
+
+//___________________________________________________________________________________________
+
+/// Performs MC matching.
+struct HFCandidateCreatorCascadeMC {
+  Produces<aod::HfCandCascadeMCRec> rowMCMatchRec;
+  Produces<aod::HfCandCascadeMCGen> rowMCMatchGen;
+
+#ifdef MY_DEBUG
+  Configurable<std::vector<int>> v_labelK0Spos{"v_labelK0Spos", {729, 2866, 4754, 5457, 6891, 7824, 9243, 9810}, "labels of K0S positive daughters, for debug"};
+  Configurable<std::vector<int>> v_labelK0Sneg{"v_labelK0Sneg", {730, 2867, 4755, 5458, 6892, 7825, 9244, 9811}, "labels of K0S negative daughters, for debug"};
+  Configurable<std::vector<int>> v_labelProton{"v_labelProton", {717, 2810, 4393, 5442, 6769, 7793, 9002, 9789}, "labels of protons, for debug"};
+#endif
+
+  void process(aod::HfCandCascade const& candidates,
+               aod::BigTracksMC const& tracks,
+               aod::McParticles const& particlesMC)
+  {
+    int8_t sign = 0;
+    std::vector<int> arrDaughLcIndex;
+    std::array<int, 3> arrDaughLcPDG;
+    std::array<int, 3> arrDaughLcPDGRef = {2212, 211, -211};
+
+    // Match reconstructed candidates.
+    for (auto& candidate : candidates) {
+      auto arrayDaughtersV0 = array{candidate.posTrack_as<aod::BigTracksMC>(), candidate.negTrack_as<aod::BigTracksMC>()};
+      auto arrayDaughtersLc = array{candidate.index0_as<aod::BigTracksMC>(), candidate.posTrack_as<aod::BigTracksMC>(), candidate.negTrack_as<aod::BigTracksMC>()};
```

Comment:
`posTrack_as` and `negTrack_as` should not be called twice. Save the track rows in variables.

Reply:
Still there

Reply:
I hope I fixed them all now.